src/common/config.cc

```diff
@@ -1202,18 +1201,19 @@ void Config::GenData() {
                     this->ofdm_ca_num_ * this->ue_ant_num_,
                     Agora_memory::Alignment_t::kAlign64);
   std::vector<FILE*> vec_fp_tx;
-  if (kOutputUlFreqData) {
-    for (size_t u = 0; u < this->ue_ant_num_; u++) {
+  if (kOutputUlScData) {
+    for (size_t i = 0; i < this->ue_num_; i++) {
       const std::string filename_ul_data_f =
           kUlDataFreqPrefix + ul_modulation_ + "_" +
           std::to_string(ofdm_data_num_) + "_" + std::to_string(ofdm_ca_num_) +
-          "_1_" + std::to_string(this->frame_.NumULSyms()) + "_1_A_" +
-          std::to_string(u) + ".bin";
+          "_" + std::to_string(kOfdmSymbolPerSlot) + "_" +
+          std::to_string(this->frame_.NumULSyms()) + "_" +
+          std::to_string(kOutputFrameNum) + "_" + ue_channel_ + "_" +
+          std::to_string(i) + ".bin";
       FILE* fp_tx_f = std::fopen(filename_ul_data_f.c_str(), "wb");
       if (fp_tx_f == nullptr) {
-        AGORA_LOG_ERROR("Failed to create antenna file %s. Error %s.\n",
+        AGORA_LOG_ERROR("Failed to create ul sc data file %s. Error %s.\n",
                         filename_ul_data_f.c_str(), strerror(errno));
-        throw std::runtime_error("Config: Failed to create ul antenna file");
```

Comment:
You must throw the error here or otherwise handle the possibility of a null pointer in the array.